### PR TITLE
Feat/sort carets

### PR DESCRIPTION
**Pull request by JeffreyStocker**
Added sorting indicators to sort buttons in tables
They can be observed by going to a page with a list table, then selecting a header to sort a table.

### CODE REVIEW

File: adventures.html [hunk 1]
```diff
@@ -55,12 +55,11 @@ <h1>Adventures</h1>
 			<input type="search" id="search" class="search form-control" placeholder="Find adventure...">
 			<span class="input-group-btn"><button class="btn btn-default" type="button" id="reset">Reset</button></span>
 		</div>
-
 		<div id="filtertools" class="sortlabel btn-group">
-			<button class="col-xs-6 sort btn btn-default btn-xs" type="button" data-sort="name">Name</button>
-			<button class="col-xs-2 col-xs-2-5 sort btn btn-default btn-xs" type="button" data-sort="storyline">Storyline</button>
-			<button class="col-xs-1 col-xs-1-3 sort btn btn-default btn-xs" type="button" data-sort="level">Levels</button>
-			<button class="col-xs-2 col-xs-2-2 sort btn btn-default btn-xs" type="button" data-sort="published">Published</button>
+			<button class="col-xs-6 sort btn btn-default btn-xs" type="button" data-sort="name">Name <span class="" \></button>
```
Comment (TheGiddyLimit) [inline adventures.html]: Use correct HTML tags: `<span></span>`

Remove empty `class` attributes

File: adventures.html [hunk 2]
```diff
@@ -55,12 +55,11 @@ <h1>Adventures</h1>
...
+			<button class="col-xs-2 col-xs-2-5 sort btn btn-default btn-xs" type="button" data-sort="storyline" >Storyline <span class=""\></button>
```
Comment (TheGiddyLimit) [inline adventures.html]: Extra space

File: backgrounds.html [hunk 3]
```diff
@@ -92,6 +92,7 @@ <h1>Backgrounds List</h1>
 <script type="text/javascript" src="js/scalecreature.js"></script>
 <script type="text/javascript" src="js/filter.js"></script>
 <script type="text/javascript" src="js/entryrender.js"></script>
+<script type="text/javascript" src="js/sortCaret.js"></script>
```
Comment (TheGiddyLimit) [inline backgrounds.html]: Add this to `utils.js` and initialise it with a function call

File: js/bestiary.js [hunk 4]
```diff
@@ -315,7 +315,11 @@ function pPageInit (loadedSources) {
 	// sorting headers
 	$("#filtertools").find("button.sort").on(EVNT_CLICK, function () {
 		const $this = $(this);
-		$this.data("sortby", $this.data("sortby") === "asc" ? "desc" : "asc");
+		var direction = $this.data("sortby") === "desc" ? "asc" : "desc"
```
Comment (TheGiddyLimit) [inline js/bestiary.js]: Use `let`/`const`

File: js/bookslist.js [hunk 5]
```diff
@@ -17,17 +17,26 @@ class BooksList {
...
+		var list;
 		this.list = new List("listcontainer", {
 			valueNames: ["name", "source"],
 			listClass: "books"
 		});
+		list = this.list;
 
-		$("#filtertools").find("button.sort").on(EVNT_CLICK, function () {
+		$("#filtertools").find("button.sort").on(EVNT_CLICK, function (evt) {
+			// console.log ('evt', evt);
```
Comment (TheGiddyLimit) [inline js/bookslist.js]: Remove test code

File: js/sortCaret.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+var $$filterButtonFunction = function (target, $this = $(this), direction) {
```
Comment (TheGiddyLimit) [inline js/sortCaret.js]: Move to `utils.js` (including in `SortUtil` would be appropriate enough)

Rename to `handleFilterButtonClick` or similar

(Use `let`/`const`)

File: js/sortCaret.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+	if (!direction) {
+		direction = $this.hasClass("asc") || $this.attr("data-sortby") === "asc" ? "asc" : "desc";
+	}
+
+	$(target).find(".caret").removeClass("caret");
+	$this.find("span").addClass("caret")
+		.toggleClass("caret-reverse", direction === "asc");
+};
+
+$("#filtertools").find("button.sort").click(function () { $$filterButtonFunction.call(this, "#filtertools"); });
```
Comment (TheGiddyLimit) [inline js/sortCaret.js]: Split `function () { $$filterButtonFunction.call(this, "#filtertools"); }` across three lines

Add missing newline at end of file

**Comment by JeffreyStocker** [comment]
Requested changes should be done, 

I placed the initialization function into a new object in utils in initializationFunction then call it in each html's windows.onload function. 
I didn't see a better util object to store the initialization function in. Let me know if there is a better place to put it.

**Comment by TheGiddyLimit** [comment]
It'll work; thanks

**Pull request merged**